src/views/Upload.vue

```diff
@@ -126,6 +153,27 @@ export default {
 						type: 'error',
 					});
 				}
+				
+				if (isEmailProvided) {
+					email = this.email; 
+
+					if (email == "") {
+						return this.$store.dispatch('AlertStore/setAlert', {
+						message:
+							'Nezadali ste emailovú adresu !',
+						type: 'error',
+						});
+					}
+				} else {
+					if (this.email !== "") {
+						return this.$store.dispatch('AlertStore/setAlert', {
```

Comment:
Actually, If the email input was disabled, then user cannot enter an email and you wouldn't need this second condition.